### ISSUE TITLE
RecursionError when doing :Denite file/rec several times

### ISSUE THREAD

**Issue by gauteh** [Shougo/denite.nvim#516]
**Warning:  I will close the issue without the minimal init.vim and the reproduction instructions.**

# Problems summary

I am getting a crash when doing `:Denite file/rec`. Usually not the first time, and it seems to sometimes be related to having opened a split window.. Either way -- this might be a recursive function thing. The full log is attached, but below is the relevant traceback extracted:

```
[denite] Traceback (most recent call last):
[denite]   File "<string>", line 9, in _temporary_scope
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/ui/default.py", line 71, in start
[denite]     self._start(context['sources_queue'][0], context)
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/ui/default.py", line 127, in _start
[denite]     self.init_buffer()
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/ui/default.py", line 151, in init_buffer
[denite]     self._prev_wininfo = self._get_wininfo()
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/ui/default.py", line 247, in _get_wininfo
[denite]     wininfo = self._vim.call('getwininfo', self._vim.call('win_getid'))[0]
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 62, in call
[denite]     return reform_bytes(self.Function(name)(*args))
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 18, in reform_bytes
[denite]     return list(map(reform_bytes, value))
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 15, in reform_bytes
[denite]     reform_bytes(k): reform_bytes(v) for k, v in value.items()
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 15, in <dictcomp>
[denite]     reform_bytes(k): reform_bytes(v) for k, v in value.items()
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 15, in reform_bytes
[denite]     reform_bytes(k): reform_bytes(v) for k, v in value.items()
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 15, in <dictcomp>
[denite]     reform_bytes(k): reform_bytes(v) for k, v in value.items()
[denite]   File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 15, in reform_bytes
.....
Repeated loads and loads of times.. see full log attached.
.....
    reform_bytes(k): reform_bytes(v) for k, v in value.items()
  File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 15, in <dictcomp>
    reform_bytes(k): reform_bytes(v) for k, v in value.items()
  File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 15, in reform_bytes
    reform_bytes(k): reform_bytes(v) for k, v in value.items()
  File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 15, in <dictcomp>
    reform_bytes(k): reform_bytes(v) for k, v in value.items()
  File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 15, in reform_bytes
    reform_bytes(k): reform_bytes(v) for k, v in value.items()
  File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 15, in <dictcomp>
    reform_bytes(k): reform_bytes(v) for k, v in value.items()
  File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 18, in reform_bytes
    return list(map(reform_bytes, value))
  File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 13, in reform_bytes
    elif isinstance(value, (dict, vim.Dictionary, vim.Options)):
RecursionError: maximum recursion depth exceeded in __instancecheck__

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 16, in <module>
  File "<string>", line 14, in _temporary_scope
  File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/util.py", line 45, in error
    vim.call('denite#util#print_error', string)
  File "/home/gaute/dev/misc/dotfiles/vim/.vim/plugged/denite.nvim/rplugin/python3/denite/rplugin.py", line 62, in call
    return reform_bytes(self.Function(name)(*args))
KeyboardInterrupt
```

## Environment Information (Required!)

 * denite version (SHA1):` e1327c1

 * OS: ArchLinux

 * Vim/neovim version:
```
VIM - Vi IMproved 8.1 (2018 May 17, compiled May 25 2018 00:50:25)
Included patches: 1-22
Compiled by Arch Linux
Huge version with GTK3 GUI.  Features included (+) or not (-):
+acl               +farsi             +mouse_sgr         -tag_any_white
+arabic            +file_in_path      -mouse_sysmouse    +tcl/dyn
+autocmd           +find_in_path      +mouse_urxvt       +termguicolors
-autoservername    +float             +mouse_xterm       +terminal
+balloon_eval      +folding           +multi_byte        +terminfo
+balloon_eval_term -footer            +multi_lang        +termresponse
+browse            +fork()            -mzscheme          +textobjects
++builtin_terms    +gettext           +netbeans_intg     +timers
+byte_offset       -hangul_input      +num64             +title
+channel           +iconv             +packages          +toolbar
+cindent           +insert_expand     +path_extra        +user_commands
+clientserver      +job               +perl/dyn          +vertsplit
+clipboard         +jumplist          +persistent_undo   +virtualedit
+cmdline_compl     +keymap            +postscript        +visual
+cmdline_hist      +lambda            +printer           +visualextra
+cmdline_info      +langmap           +profile           +viminfo
+comments          +libcall           +python/dyn        +vreplace
+conceal           +linebreak         +python3/dyn       +wildignore
+cryptv            +lispindent        +quickfix          +wildmenu
+cscope            +listcmds          +reltime           +windows
+cursorbind        +localmap          +rightleft         +writebackup
+cursorshape       +lua/dyn           +ruby/dyn          +X11
+dialog_con_gui    +menu              +scrollbind        -xfontset
+diff              +mksession         +signs             +xim
+digraphs          +modify_fname      +smartindent       -xpm
+dnd               +mouse             +startuptime       +xsmp_interact
-ebcdic            +mouseshape        +statusline        +xterm_clipboard
+emacs_tags        +mouse_dec         -sun_workshop      -xterm_save
+eval              +mouse_gpm         +syntax            
+ex_extra          -mouse_jsbterm     +tag_binary        
+extra_search      +mouse_netterm     +tag_old_static    
   system vimrc file: "/etc/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "/etc/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/libdrm -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -D_FORTIFY_SOURCE=2  -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linking: gcc   -L. -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -fstack-protector -rdynamic -Wl,-export-dynamic -Wl,-E -Wl,-rpath,/usr/lib/perl5/5.26/core_perl/CORE  -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -L/usr/local/lib -Wl,--as-needed -o vim   -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lfribidi -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lSM -lICE -lXt -lX11 -lXdmcp -lSM -lICE  -lm -ltinfo -lelf -lnsl    -lacl -lattr -lgpm -ldl   -Wl,-E -Wl,-rpath,/usr/lib/perl5/5.26/core_perl/CORE -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -fstack-protector-strong -L/usr/local/lib  -L/usr/lib/perl5/5.26/core_perl/CORE -lperl -lpthread -lnsl -ldl -lm -lcrypt -lutil -lc   -L/usr/lib -ltclstub8.6 -ldl -lz -lpthread -lm     

```

 * `:checkhealth` or `:CheckHealth` result(neovim only):

## Provide a minimal init.vim with less than 50 lines (Required!)
.....

## How to reproduce problems from neovim startup (Required!)

Doesn't always happen, but stress testing `:Denite file/rec` in my .vim folder with a few files open seems to do the trick.

## Generate a logfile if appropriate

 1. vim -Vlog.vim
[vimlog.txt](https://github.com/Shougo/denite.nvim/files/2260109/vimlog.txt)

**Comment by Shougo** [comment]
Your Vim version is old.
Please use the latest version.

I think it is not reproduced in neovim.

**Comment by Shougo** [comment]
> Doesn't always happen, but stress testing :Denite file/rec in my .vim folder with a few files open seems to do the trick.

I cannot reproduce the problem.
Please describe the reproduce instructions more clearly.

If you don't provide it, I will close the issue unfortunately.

**Comment by gauteh** [comment]
Shougo writes on August 6, 2018 4:41:
> Your Vim version is old.
> Please use the latest version.

I do not think it is very old (released this May), its the latest 
available in Arch Linux.

> I think it is not reproduced in neovim.

Have not tried. But it is supposed to work on Vim 8 as well?

I haven't been able to reliably reproduce the issue, but it happens 
often enough that I cannot use it. I understand if you close the issue, 
but unfortunately this makes the plugin unusable for me.

**Comment by Shougo** [comment]
> I do not think it is very old (released this May), its the latest available in Arch Linux.

It is old.  Because, 8.1.240 is already released.
https://github.com/vim/vim/commit/3cb4448b8a5c0192988f4e349aba6d7a91a9a4bd

>  But it is supposed to work on Vim 8 as well? 

It works, but it uses compatible plugin(rplugin.py).
It may have problems.  neovim is better.

> I haven't been able to reliably reproduce the issue, but it happens often enough that I cannot use it. I understand if you close the issue, but unfortunately this makes the plugin unusable for me.

Of course, I understand.
But to fix the problem, denite must use nvim-yarp instead.
It adds the another dependency.
So I don't want to do it.

**Comment by Shougo** [comment]
Please built Vim8.1 HEAD manually.
I don't recommend for you to use ArchLinux released version.
It is very old for me.

**Comment by gauteh** [comment]
Ok, that is not an option for me unfortunately since I need my vim setup to work on several machines. Some of them on slightly older systems where I do not have the possibility to run the latest vim. Was this fixed in  27ffda7? Otherwise, I understand your decision.

**Comment by Shougo** [comment]
> Was this fixed in 27ffda7?

Yes.
